fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -731,6 +736,61 @@ where
     }
 }
 
+impl<M, S, Tx, Ecal> Interpreter<M, S, Tx, Ecal>
+where
+    S: InterpreterStorage,
+{
+    fn blob_inner(
+        blob: &mut Blob,
+        storage: &mut S,
+        initial_balances: InitialBalances,
+        gas_costs: &GasCosts,
+        fee_params: &FeeParameters,
+        base_asset_id: &AssetId,
+        gas_price: Word,
+    ) -> Result<(), InterpreterError<S::DataError>> {
+        let blob_data = blob
+            .witnesses()
+            .get(*blob.bytecode_witness_index() as usize)
+            .ok_or(InterpreterError::Bug(Bug::new(
+                // It shouldn't be possible since `Checked<Blob>` guarantees
+                // the existence of the witness.
+                BugVariant::WitnessIndexOutOfBounds,
+            )))?;
+
+        let blob_id = blob.blob_id();
+
+        debug_assert_eq!(
+            BlobId::compute(blob_data.as_ref()),
+            *blob_id,
+            "Tx has invalid BlobId",
+        );
+
+        let old = storage
+            .storage_as_mut::<BlobData>()
+            .replace(blob_id, blob_data.as_ref())
+            .map_err(RuntimeError::Storage)?;
+
+        if old.is_some() {
+            return Err(InterpreterError::Panic(PanicReason::BlobIdAlreadyUploaded));
+        }
```

Comment:
Maybe it will be cheaper to use `contains` + `insert` instead of `replace`. But it can be done in a separate PR/Issue, because other places have the same problem.

Reply:
Maybe? Also should be profiled if we're doing anything for performance reasons.